examples/miral-shell/shell_main.cpp

```diff
@@ -68,9 +71,24 @@ int main(int argc, char const* argv[])
 
     SpinnerSplash spinner;
     InternalClientLauncher launcher;
+    auto constexpr default_focus_stealing = "allow";
+    auto const string_to_focus_stealing = [](std::string const& option)
+    {
+        if (option == "allow")
+        {
+            return FocusStealing::allow;
+        }
+        else if (option == "prevent")
+        {
+            return FocusStealing::prevent;
+        }
+
+        BOOST_THROW_EXCEPTION(std::runtime_error("Invalid focus stealing choice: " + option));
+    };
+    auto focus_stealing_prevention = string_to_focus_stealing(default_focus_stealing);
```

Comment:
You get better diagnostics by reporting the error by throwing `AbnormalExit`.

But could avoid the custom parsing and error reporting using a boolean option (`--focus-stealing-prevention=off`).